proxy/http2/Http2ClientSession.cc

```diff
@@ -326,6 +327,9 @@ Http2ClientSession::main_event_handler(int event, void *edata)
     break;
 
   case VC_EVENT_WRITE_COMPLETE:
+    if (this->get_half_close_flag()) {
```

Comment:
When the GOAWAY frame is sent, the sender should complete the processing of outstanding streams which are less than the stream identifier sent in the goaway frame. It seems we are not doing that here.